### DISCUSSION TITLE
HomeAssistant + Zoraxy = wrong geolocation

### DISCUSSION THREAD

**Discussion by julien-berthier** [tobychui/zoraxy#212] [General]
Good morning,

first of all, congratulations on this brilliant project!

I've been using HomeAssistant for a while, and the geolocation worked well.

Since I've been using Zoraxy, the location shows my house, not my phone.

I think it's linked to the geoip, can I add an http header to change the ip address shown?

thanks :)

**Comment by tobychui** [comment]
@julien-berthier I think that is expected behavior as Zoraxy is in your server in your house it is proxying your phone's request. So HA see the request is coming from your server in your house but not your phone. 
I am not sure what you mean by ip address in HTTP header. There are a lot of ways HA can detect your ip address like `X-Forwarded-For` or `X-Real-IP`. You can change all the headers you need in the `Custom Header` button under the edit menu of the HTTP proxy rule.
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/6f11c1ee-150d-42c0-82d0-ea2e0efa664f)

![圖片](https://github.com/tobychui/zoraxy/assets/24617523/194687d7-3ce0-4e07-a9ab-6d0bf352117c)

If you have questions on what headers needed to be changed to make HA work with reverse proxy, go ask the HA community instead.

**Reply by julien-berthier** [reply]
I also had problems with my personal apache, but it's been fixed with the right module.

Thanks you :)
